app/models/wpcc/legal_period_percents.rb

```diff
@@ -0,0 +1,61 @@
+module Wpcc
+  class LegalPeriodPercents
```

Comment:
again in reference to the naming of this...
This object is all to do with the 'legal percents' of the instantiated 'period', right? so to me it would make more sense to be named `PeriodLegalPercents`.. wdyt?